streambed-logged/src/lib.rs

```diff
@@ -945,4 +1050,113 @@ mod tests {
         assert!(records.next().await.is_some());
         assert!(records.next().await.is_none());
     }
+
+    #[test(tokio::test)]
+    async fn test_recovery() {
```

Comment:
Managed to create a comprehensive test where I write 3 records out, corrupt the 3rd one, restart the commit log to invoke recovery (verified via the debugger and by inspecting the file), and then producing a fourth and consuming the second and fourth.